src/clr-angular/data/datagrid/render/header-renderer.ts

```diff
@@ -4,16 +4,33 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { Directive, ElementRef, EventEmitter, OnDestroy, Output, Renderer2 } from '@angular/core';
-import { Subscription } from 'rxjs';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 import { DomAdapter } from '../../../utils/dom-adapter/dom-adapter';
 import { DatagridRenderStep } from '../enums/render-step.enum';
 import { ColumnResizerService } from '../providers/column-resizer.service';
 import { STRICT_WIDTH_CLASS } from './constants';
 import { DatagridRenderOrganizer } from './render-organizer';
+import { DatagridColumnState } from '../interfaces/column-state.interface';
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
 
 @Directive({ selector: 'clr-dg-column', providers: [ColumnResizerService] })
 export class DatagridHeaderRenderer implements OnDestroy {
+  stateSubscription: Subscription;
+  _columnState: BehaviorSubject<DatagridColumnState>;
+
+  get columnState(): BehaviorSubject<DatagridColumnState> {
+    return this._columnState;
+  }
+
+  set columnState(columnState: BehaviorSubject<DatagridColumnState>) {
+    if (this.stateSubscription) {
+      this.stateSubscription.unsubscribe();
+    }
+    this._columnState = columnState;
+    this.stateSubscription = columnState.asObservable().subscribe(state => this.stateChanges(state));
```

Comment:
Would it be more natural to add a getter to the columnsState instead of directly exposing the BehaviorSubject? Same comment on `cell-renderer`

Reply:
I'm not sure I understand, `columnState` is a BehaviorSubject. This is to autosubscribe to it anytime the state observable is passed back down into the cell (like if a column was hidden).

Reply:
That makes sense when I see that columnsState comes from the array of columnService.columns.